### DISCUSSION TITLE
51. IP 단편화에 대해서 설명해주세요.

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#53] [General]
`키워드`: IP의 기능, IP 단편화, MTU, 경로 MTU 발견(Path MTU Discovery)

**Comment by cutehumanS2** [comment]
IP 단편화는 전송하고자 하는 패킷의 크기가 MTU를 초과하는 경우, 이를 MTU 이하의 여러 패킷으로 나누는 것을 의미합니다.

데이터 링크 계층의 MTU는 네트워크마다 다를 수 있기 때문에, IP 패킷의 크기가 MTU보다 클 경우 패킷이 전달되지 못할 수 있습니다. 이를 해결하기 위해 IP 단편화를 수행하며, 수신 측에서는 재조립을 원래 데이터로 복구합니다. 

그러나, IP 단편화는 성능 저하와 오버헤드 증가와 같은 문제를 유발할 수 있어, 가급적 피하는 것이 좋습니다. 이를 위해 오늘날 “경로 MTU 발견(Path MTU Discovery)”과 같은 기법이 사용되기도 합니다.
+) MTU: Maximum Transmission Unit, 한 번에 전송 가능한 IP 패킷의 최대 크기

<br><br>

[참고한 자료]
<혼자 공부하는 네트워크> 03-1. LAN을 넘어서는 네트워크 계층
